menpofit/dpm/cpp/qp_one_sparse.h

```diff
@@ -0,0 +1,151 @@
+#include <math.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <iostream>
+using namespace std;
+
+#define MAX(A,B) ((A) < (B) ? (B) : (A))
+#define MIN(A,B) ((A) > (B) ? (B) : (A))
+#define INDEX(X,I) (*(int *)(X + I*mm + m))
+int n, m, mm;
+
+// Comparison function for sorting function in sumAlpha function
+static inline void setMN(int M, int N) {
+    m = M;
+    n = N;
+}
+
+
+int comp(const void *a, const void *b) {
+  return memcmp((int32_t *)a,(int32_t *)b,m*sizeof(int32_t));
```

Comment:
The casting of `a` and `b` here is unnecessary since `memcmp` expects `const void *` anyway. All that matters is the size.